ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryModule.java

```diff
@@ -53,262 +52,267 @@
+     */
+    public static ImageGalleryController getController() throws TskCoreException {
         synchronized (controllerLock) {
             if (controller == null) {
-                controller = new ImageGalleryController(Case.getCurrentCaseThrows());
+                try {
+                    Case currentCase = Case.getCurrentCaseThrows();
+                    controller = new ImageGalleryController(currentCase);
+                } catch (NoCurrentCaseException ex) {
+                    throw new TskCoreException("Failed to get ", ex);
+                }
             }
             return controller;
         }
     }
 
     /**
-     *
-     *
-     * This method is invoked by virtue of the OnStart annotation on the OnStart
-     * class class
+     * Sets the implicit exit property attribute of the JavaFX runtime to false
+     * and sets up listeners for application events. It is invoked at
+     * application start up by virtue of the OnStart annotation on the OnStart
+     * class in this package.
      */
     static void onStart() {
         Platform.setImplicitExit(false);
-        logger.info("Setting up ImageGallery listeners"); //NON-NLS
-
         IngestManager.getInstance().addIngestJobEventListener(new IngestJobEventListener());
         IngestManager.getInstance().addIngestModuleEventListener(new IngestModuleEventListener());
-        Case.addPropertyChangeListener(new CaseEventListener());
+        Case.addEventTypeSubscriber(CASE_EVENTS_OF_INTEREST, new CaseEventListener());
     }
 
+    /**
+     * Gets the image gallery module name.
+     *
+     * @return The module name,
+     */
     static String getModuleName() {
         return MODULE_NAME;
     }
 
     /**
-     * get the Path to the Case's ImageGallery ModuleOutput subfolder; ie
-     * ".../[CaseName]/ModuleOutput/Image Gallery/"
+     * Gets the path to the image gallery module output folder for a given case.
      *
-     * @param theCase the case to get the ImageGallery ModuleOutput subfolder
-     *                for
+     * @param theCase The case.
      *
-     * @return the Path to the ModuleOuput subfolder for Image Gallery
+     * @return The path to the image gallery module output folder for the case.
      */
     public static Path getModuleOutputDir(Case theCase) {
         return Paths.get(theCase.getModuleDirectory(), getModuleName());
     }
 
-    /** provides static utilities, can not be instantiated */
+    /**
+     * Prevents instantiation.
+     */
     private ImageGalleryModule() {
     }
 
-    /** is listening enabled for the given case
+    /**
+     * Indicates whether or not the image gallery module is enabled for a given
+     * case.
      *
-     * @param c
+     * @param theCase The case.
      *
-     * @return true if listening is enabled for the given case, false otherwise
+     * @return True or false.
      */
-    static boolean isEnabledforCase(Case c) {
-        if (c != null) {
```

Comment:
Sure, I can do that. I wonder if the NetBeans IDE does any analysis so that the annotation is more than an assertion...

Note that the point in me removing the null check of the parameter is that this method is not public API and passing a null Case reference is a programming error that I want to have pop out as an NPE, rather than be silently converted into a false return value.